hazelcast/src/test/java/com/hazelcast/client/impl/spi/impl/discovery/ViridianAddressProviderTest.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.spi.impl.discovery;
+
+import com.hazelcast.client.impl.management.ClientConnectionProcessListenerRunner;
+import com.hazelcast.cluster.Address;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category(QuickTest.class)
+public class ViridianAddressProviderTest {
+
+    private static final Address PRIVATE_MEMBER_ADDRESS = Address.createUnresolvedAddress("10.0.0.1", 30000);
+    private static final Address PUBLIC_MEMBER_ADDRESS = Address.createUnresolvedAddress("34.215.70.2", 32661);
+
+    private static final Address PRIVATE_TPC_ADDRESS = Address.createUnresolvedAddress("10.0.0.1", 42000);
+    private static final Address PUBLIC_TPC_ADDRESS = Address.createUnresolvedAddress("34.215.70.2", 52553);
+    private static final Address NON_EXISTENT_PRIVATE_ADDRESS = Address.createUnresolvedAddress("1.1.1.1", 1111);
+
+    private final Map<Address, Address> privateToPublic = new HashMap<>();
+    private final List<Address> members = new ArrayList<>();
+
+    @Test
+    public void testLoadAddresses() throws Exception {
+        setUp();
+        ViridianAddressProvider provider = new ViridianAddressProvider(createDiscovery());
+        Collection<Address> addresses = provider.loadAddresses(createListenerRunner()).primary();
```

Comment:
maybe we can assert that listener runner's possibleAddressesCollected method is called as well? also maybe we can assert the secondary() as well.

Reply:
https://github.com/hazelcast/hazelcast/pull/24378/commits/254ac37b5d003322c57e47ce8d5ac6d0ad7fe7be

Reply:
I didn't get the part about the `also maybe we can assert the secondary() as well.` though. What did you mean there?

Reply:
I mean the public addresses (Addresses.secondary())

Reply:
OK, got it https://github.com/hazelcast/hazelcast/pull/24378/commits/593f37b478e7b25a71b682506629d340b04aedbb